src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -433,8 +452,15 @@ public final void run() {
                         // in an immediately ready selection key, hence we use the non-blocking form
                         selected = selector.selectNow();
                     } else {
-                        selected = selector.select(); // WINDOWS!!! Waiting the whole timeout anyway if timeout specified?!?
+                        // On Windows the select(timeout) operation ALWAYS waits for the timeout,
+                        // so we workaround it by WindowsIOHubSelectorWatcher
```

Comment:
as part of this can you file a JDK bug with a reproducable scenario - just to make sure we are not doing something slightly wrong...